Makefile variables need quotes to properly handle system paths with spaces (Windows/WSL2 + Docker)

### Description of Issue

Building firmware on the V3.0 branch at commit 1728a660cea95760573a72bbf0f6b1da68342749 `Prefer tr to ${char^^}, which does not work on older bash versions (#303)`.

Building on Windows/WSL2 with Docker:

```
$make
```
Produces the error:
```
/mnt/c/Program Files/Docker/Docker/resources/bin/docker build --tag zmk --file Dockerfile .
make: /mnt/c/Program: No such file or directory
make: *** [Makefile:17: all] Error 127
```

The issue is that the docker path has spaces:

```
$command -v docker
```
Returns the output
```
/mnt/c/Program Files/Docker/Docker/resources/bin/docker
```

This only seems to happen when Docker is in "Resource Saver Mode". When the docker engine is running, the command returns
```
$command -v docker
/usr/bin/docker
```

### Suggested Fix

Recommended to wrap shell variables in quotation marks.

```diff
diff --git a/Makefile b/Makefile
index e5231eb..4b31591 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
-DOCKER := $(shell { command -v podman || command -v docker; })
-TIMESTAMP := $(shell date -u +"%Y%m%d%H%M")
-COMMIT := $(shell git rev-parse --short HEAD 2>/dev/null)
-detected_OS := $(shell uname)  # Classify UNIX OS
+DOCKER := "$(shell { command -v podman || command -v docker; })"
+TIMESTAMP := "$(shell date -u +"%Y%m%d%H%M")"
+COMMIT := "$(shell git rev-parse --short HEAD 2>/dev/null)"
+detected_OS := "$(shell uname)" # Classify UNIX OS
 ifeq ($(strip $(detected_OS)),Darwin) #We only care if it's OS X
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@ Here's all notable changes and commits to both the configuration repo and the ba
 Many thanks to all those who have submitted issues and pull requests to make this firmware better!
 ## Config repo
 
+1/16/2024 - Change the makefile to fis WSL2 compatibility [#335](https://github.com/KinesisCorporation/Adv360-Pro-ZMK/pull/335)
+
 1/14/2024 - Update base ZMK, change KConfig attributes to support, Enable experimental BLE features for improved stability [#326](https://github.com/KinesisCorporation/Adv360-Pro-ZMK/pull/326)
 
 12/27/2023 - Change how the characters are used in the versioning script for improved MacOS experience [#303](https://github.com/KinesisCorporation/Adv360-Pro-ZMK/pull/303)

--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
-DOCKER := $(shell { command -v podman || command -v docker; })
-TIMESTAMP := $(shell date -u +"%Y%m%d%H%M")
-COMMIT := $(shell git rev-parse --short HEAD 2>/dev/null)
-detected_OS := $(shell uname)  # Classify UNIX OS
+DOCKER := "$(shell { command -v podman || command -v docker; })"
+TIMESTAMP := "$(shell date -u +"%Y%m%d%H%M")"
+COMMIT := "$(shell git rev-parse --short HEAD 2>/dev/null)"
+detected_OS := "$(shell uname)"  # Classify UNIX OS
 ifeq ($(strip $(detected_OS)),Darwin) #We only care if it's OS X
 SELINUX1 :=
 SELINUX2 :=

